### ISSUE TITLE
Zip archive reading fails for empty file

### ISSUE THREAD

**Issue by ymzong** [autolab/Autolab#192]
The following exception happens when a file inside a zip archive is empty:

```
A NoMethodError occurred in background at 2015-01-26 18:35:17 -0500 :

  undefined method `read' for Zip::NullInputStream:Module
  /opt/autolab/build/Autolab/app/controllers/submissions_controller.rb:791:in `block in getFileAt'
```

**Comment by dlbucci** [comment]
This can happen if there's a folder inside of the zip file and someone tries to download it.  I think our old archive code handled nested files, but our new one appears to just show the top level folder...

**Comment by dlbucci** [comment]
Fixed by #357
